lte/gateway/release/pydep

```diff
@@ -888,54 +1008,56 @@ def main(args):
-                            existing_versions[key] = ver
-                        except KeyError as e:
-                            log.error('{} missing key: {}'.format(key, e))
+                lf = Lockfile(data=r.read())
         except FileNotFoundError:
             pass
+    else:
+        lf = Lockfile()
+
+    deps = lf.dependencies()
+    for key in deps:
+        if deps[key]['source'] == 'pypi':
+            try:
+                ver = typing.cast(str, deps[key]['version'])
+                existing_versions[key] = ver
+            except KeyError as e:
+                log.error('{} missing key: {}'.format(key, e))
 
-        log.debug(deps)
+    log.debug(deps)
 
     repo_pkgs = [gen_sys_package_name(p.key, args.use_py2) for p in repo_installable]
     if args.install_from_repo:
         try:
-            log.error(repo_pkgs)
             subprocess.call(shlex.split('sudo apt install -y '
                                         + ' '.join(repo_pkgs)))
         except Exception:
             log.error('error trying to install repo packages')
             raise
 
-    pip_distributions = gen_pip_distributions(root_requirements,
-                                              existing_versions=existing_versions,
-                                              pypi_only=args.force_pypi,
-                                              py2=args.use_py2)
-
-    required_distributions = {}
-    to_traverse = [pip_distributions[k] for k in [req.key for req in input_root_requirements]]
-    while to_traverse:
-        dist = to_traverse.pop()
-        required_distributions[dist.key] = dist
-        requires = dist.requires()
-        log.debug(requires)
-        for req in requires:
-            prereq_dist = pip_distributions.get(req.key)
-            if prereq_dist:
-                to_traverse.append(prereq_dist)
-            else:
-                log.error('{} required but not detected'.format(prereq_dist))
+    with tmpvirtualenv() as venv:
+        pip_distributions = gen_pip_distributions(root_requirements,
+                                                  existing_versions=existing_versions,
+                                                  pypi_only=args.force_pypi,
+                                                  py2=args.use_py2,
+                                                  venv=venv)
+
+        required_distributions = {}
+        to_traverse = [pip_distributions[k] for k in [req.key for req in input_root_requirements]
+                       if k not in repo_installable]
+        while to_traverse:
+            dist = to_traverse.pop()
+            required_distributions[dist.key] = dist
+            requires = dist.requires()
+            for req in requires:
+                prereq_dist = pip_distributions.get(req.key)
+                if prereq_dist:
+                    to_traverse.append(prereq_dist)
+                else:
+                    log.error('{} required but not detected'.format(prereq_dist))
```

Comment:
should `prereq_dist` being logged here? it seems here we will log an error if this variable is `None` so not sure why this should be appended to the log

Reply:
I will send fix in separate PR.